packages/snaps-execution-environments/src/common/endowments/index.ts

```diff
@@ -52,6 +52,10 @@ export function createEndowments(
   ethereum: StreamProvider,
   endowments: string[] = [],
 ): { endowments: Record<string, unknown>; teardown: () => Promise<void> } {
+  // Harden custom endowments
+  harden(snap);
```

Comment:
Could we just harden these where they are returned instead? Just seems a bit more clear

Reply:
Yes, I've added changes matching your request.